lib/phoenix/controller.ex

```diff
@@ -894,6 +894,11 @@ defmodule Phoenix.Controller do
     * `:content_type` - the content type of the file or binary
       sent as download. It is automatically inferred from the
       filename extension
+    * `:disposition_type` - specifies dispositon type
+      (`"attachment"` or `"inline"`). If `"attachment"` was used,
+      user will be prompted to save the file. If `"inline"` was used,
+      the browser will attempt to open the file.
+      Defaults to `"attachment"`.
```

Comment:
Let's make it an atom and let's assert it is either the atoms `:attachment` or `:inline`. That's because generally allowing strings can lead to vulnerability.

Also, I would say `:disposition` is good enough for the option name, no need for the `_type` suffix. Thoughts?

Reply:
Yes, I was thinking that the value of this option should be somehow constrained. Any suggestions how to do it?
And yes, changing option name to just `:disposition` makes sense :)

Reply:
> Yes, I was thinking that the value of this option should be somehow constrained. Any suggestions how to do it?

Something like this:

```elixir
case Keyword.get(options, :disposition, :attachment) do
  :attachment -> "attachment"
  :inline -> "inline"
  other -> raise ArgumentError, "expected :disposition to be :attachment or :inline, got: #{inspect(other)}"
end
```

Reply:
Thanks @josevalim
I'll do it later today.

Reply:
The name `:disposition_type` was chosen to reflect the naming of this item in RFC describing Content-disposition header. https://tools.ietf.org/html/rfc6266#section-4.2l

RFC does not constrain possible values either, just says that anything other than "inline" or "attachment" should be handled as "attachment".

But I am okay with all the suggested changes, just explaining why the original parameter anme and choice of string was made.

Reply:
@josevalim I've made changes you suggested.